pkgs/by-name/in/inochi-session/package.nix

```diff
@@ -0,0 +1,159 @@
+    rev = "49bb5ce65f7d5eeab7861d8ffd5aa2a58ca8f08c";
+    hash = "sha256-XcnZbIjwq7vmYBnMAs+cEpJL8HB8wrL098FXGxC+diA=";
+    fetchSubmodules = true;
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "inochi-session";
+  version = "0.8.3";
+
+  src = fetchFromGitHub {
+    owner = "Inochi2D";
+    repo = "inochi-session";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-yq/uMWEeydZun07/7hgUaAw3IruRqrDuGgbe5NzNYxw=";
+  };
+
+  nativeBuildInputs = [
+    dub
+    ldc
+    cmake # used for building `i2d-imgui`
+    copyDesktopItems
+    makeWrapper
+
+    # A fake git implementation for the `gitver` package used by the main package and `inochi2d`
+    # This will cause inochi2d to have the same version as the main package, thought that should already be the case
+    (writeShellScriptBin "git" "echo ${finalAttrs.src.rev}")
+  ];
+
+  buildInputs = [
+    dbus
+    freetype
+    SDL2
+  ];
+
+
+  postUnpack = ''
+    mkdir dub-deps
+    export DEPS=$(realpath dub-deps)
+    ${builtins.concatStringsSep "\n" (map (dep: ''
+      cp -r --no-preserve=all ${dep.src} $DEPS/${dep.pname}
+    '') deps)}
+  '';
+
+  postPatch = ''
+    # As per request of the author (https://github.com/NixOS/nixpkgs/issues/170315)
+    # we are not allowed to package the logo and the mascot of the application
+    sed -i '/InBranding/d' dub.sdl
+
+    sed -i "/bindbc-spout2/d" dub.sdl # windows-only dependency
+    sed -i "/vibe-d/d" $DEPS/facetrack-d/dub.sdl # behind the "web-adaptors" configuration of `facetrack-d`
+    sed -i "/dunit/d" $DEPS/ddbus/dub.json # would be used by the tests of `ddbus`
+
+    cp -r --no-preserve=all ${cimgui-src}/* $DEPS/i2d-imgui/deps/cimgui
+    sed -i $DEPS/i2d-imgui/deps/CMakeLists.txt \
+        -e "/PullSubmodules/d" \
+        -e "s/\''${cimgui_SUBMOD_DIR}/cimgui/g"
+  '';
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+
+    # register downloaded dependencies
+    ${builtins.concatStringsSep "\n" (map (dep: ''
+      dub add-local $DEPS/${dep.pname} ${dep.version}
+    '') deps)}
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    dub build --skip-registry=all --build=release --config=meta # uses the fake git
+    dub build --skip-registry=all --build=release --config=linux-full
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 LICENSE out/*LICENSE* -t $out/share/doc
+    install -Dm755 out/*.so -t $out/lib
+    install -Dm755 out/inochi-session -t $out/bin
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "inochi-session";
+      exec = "inochi-session";
+      desktopName = "Inochi Session";
+      comment = finalAttrs.meta.description;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  preFixup = ''
+    patchelf $out/bin/inochi-session --add-needed cimgui.so
+  '';
+
+  dontStrip = true; # symbol lookup error: undefined symbol: , version
```

Comment:
Your NEEDED modification causes this I think. I'm not an expert on this but what fixes this is to add the exact so path to NEEDED.

This could perhaps also work if you additionally added cimgui to the rpath.

Reply:
The second also works and is probably cleaner. See https://github.com/NixOS/nixpkgs/pull/281753/commits/3f89316af000885845d34c9a6776d8344af6a4c9 for instance.

Reply:
No matter what I seem to do, I cannot get this to work with any of the methods you listed. With `--add-rpath $out/lib` it can't even find `libc`.